app/controllers/legacy/sessions_controller.rb

```diff
@@ -285,6 +288,22 @@ def email_usernames
 
     private #################
 
+    def save_new_params_in_session
+      # Store these params in the session so they are available if the lookup_login
+      # fails.  Also these methods perform checks on the alternate signup URL.
+      set_client_app(params[:client_id])
+
+      set_alternate_signup_url(params[:signup_at])
+
+      set_student_signup_role(params[:go] == 'student_signup')
+    end
+
+    def maybe_skip_to_sign_up
+      if %w{signup student_signup}.include?(params[:go])
+        redirect_to signup_path(bpff: 9)
```

Comment:
what is the magic number of 9? consider a constant to document the intention?

Reply:
On top some refactoring I did, added this comment: 
`# — Intentionally made a little obscure. There is no magic to the number nine, just a truthy value.`